### PR TITLE
Cleaning up localization docs

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: aspnet/fundamentals/localization.rst [hunk 1]
```diff
@@ -27,7 +27,7 @@ Introduced in ASP.NET Core, `IStringLocalizer <https://docs.asp.net/projects/api
...
+In the code above, the ``IStringLocalizer<T>`` implementation comes from :doc:`/fundamentals/dependency-injection`. If the localized value of "About Title" is not found, then the indexer key is returned, that is, the string "About Title". You can leave the default language literal strings in the app and wrap them in the localizer, so that you can focus on developing the app. You develop your app with your default language and prepare it for the localization step without first creating a default resource file. Alternatively, you can use the traditional approach and provide a key to retrieve the default language string. For many developers the new workflow of not having a default language *.resx* file and simply wrapping the string literals can reduce the overhead of localizing an app. Other developers will prefer the traditional work flow as it can make it easier to work with longer string literals and make it easier to update localized strings.
```
Comment (pranavkm) [inline aspnet/fundamentals/localization.rst]: I removed the line 

>  I'll show how the ``IStringLocalizer`` service gets added in the **Configuring localization** section. 

The rest of stays as is.

File: aspnet/fundamentals/localization.rst [hunk 2]
```diff
@@ -119,8 +119,6 @@ ASP.NET Core allows you to specify two culture values, ``SupportedCultures`` and
   :local:
   :depth: 1
   
-.. note:: Currently, resource files are not read when the project is run from Visual Studio. See `this issue <https://github.com/aspnet/dnx/issues/3047>`_ for more information. Until the issue with Visual Studio is addressed, you can test the project by running it from the command line.
```
Comment (pranavkm) [inline aspnet/fundamentals/localization.rst]: This is no longer valid(?) per @DamianEdwards

**Comment by danroth27** [comment]
@pranavkm Are these changes RTM specific? If yes, then this content will need to go into dev. Master is for RC2 content.

**Comment by pranavkm** [comment]
Tthe note stopped applying since we moved to dotnet and the remainder are just cleanup - so it does apply to RC2. Does it make sense to send it to master in this case?

**Comment by Rick-Anderson** [comment]
looks good.

File: aspnet/fundamentals/localization.rst (hunk 2 above)
Comment (Rick-Anderson) [inline aspnet/fundamentals/localization.rst]: Can we remove this and apply the PR to dev?

**Pull request merged**